tests/api.rs

```diff
@@ -495,6 +495,55 @@ mod tests {
         )
     }
 
+    #[test]
```

Comment:
I appreciate you adding tests (thanks!), but just noting here that `tests/api.rs` is currently commented out in master, due to an issue with forked test and the recent Tokio upgrade.

This is being tracked in #6984.

I think it's good to leave these in and just comment out the whole file, so when #6984 is closed, these new tests are inherited.